src/IW/Db/Issue.hs

```diff
@@ -14,32 +14,35 @@ import IW.Core.Issue (Issue (..))
 import IW.Core.Id (Id (..))
 import IW.Db.Functions (WithDb, asSingleRow, executeMany, query, queryRaw)
 
+
 -- | Returns all issues in the database
 getIssues :: (WithDb env m) => m [Issue]
 getIssues = queryRaw [sql|
-    SELECT id, number, title, body, repo_owner, repo_name, labels
+    SELECT id, repo_owner, repo_name, number, title, body, labels
     FROM issues
 |]
 
 -- | Returns a single issue with the corresponding ID
 getIssueById :: (WithDb env m, WithError m) => Id Issue -> m Issue
 getIssueById issueId = asSingleRow $ query [sql|
-    SELECT id, number, title, body, repo_owner, repo_name, labels
+    SELECT id, repo_owner, repo_name, number, title, body, labels
     FROM issues
     WHERE id = ?
 |] (Only issueId)
 
 -- | Returns a list of issues filtered by label name
 getIssuesByLabel :: (WithDb env m) => Text -> m [Issue]
 getIssuesByLabel label = query [sql|
-    SELECT id, number, title, body, repo_owner, repo_name, labels 
+    SELECT id, repo_owner, repo_name, number, title, body, labels
     FROM issues 
     WHERE ? = ANY (labels)
 |] (Only label)
 
--- | Insert a list of issues into the database
-insertIssues :: (WithDb env m) => [Issue] -> m ()
-insertIssues issues = executeMany [sql|
-    INSERT INTO issues (number, title, body, repo_owner, repo_name, url, labels)
-    VALUES (?, ?, ?, ?, ?, ?, ?)
+-- | Insert a list of issues into the database, but update on conflict
+upsertIssues :: (WithDb env m) => [Issue] -> m ()
+upsertIssues issues = executeMany [sql|
+    INSERT INTO issues (repo_owner, repo_name, number, title, body, labels)
+    VALUES (?, ?, ?, ?, ?, ?)
+    ON CONFLICT (repo_owner, repo_name, number) DO 
+    UPDATE SET title = EXCLUDED.title, body = EXCLUDED.body, labels = EXCLUDED.labels;
```

Comment:
Let's format this SQL query a bit differently to improve readability:

```sql
    INSERT INTO issues 
      (repo_owner, repo_name, number, title, body, labels)
    VALUES
      (?, ?, ?, ?, ?, ?, ?)
    ON CONFLICT (repo_owner, repo_name, number) DO 
    UPDATE SET
        title = EXCLUDED.title
      , body = EXCLUDED.body
      , labels = EXCLUDED.labels
```

Reply:
This is what the query looks like: 
```
    INSERT INTO issues (repo_owner, repo_name, number, title, body, labels)
    SELECT repo_owner, repo_name, number, title, body, labels
    FROM (VALUES (?, ?, ?, ?, ?, ?))
    AS new (repo_owner, repo_name, number, title, body, labels)
    WHERE EXISTS (
        SELECT (owner, name) FROM repos WHERE (repos.owner, repos.name) = (new.repo_owner, new.repo_name)
    )
    ON CONFLICT ON CONSTRAINT unique_issues DO 
    UPDATE SET 
        title = EXCLUDED.title
      , body = EXCLUDED.body
      , labels = EXCLUDED.labels;
```
Is this fine?

Reply:
```
    INSERT INTO issues 
        (repo_owner, repo_name, number, title, body, labels)
    SELECT 
        repo_owner, repo_name, number, title, body, labels
    FROM (
        VALUES 
            (?, ?, ?, ?, ?, ?)
         )
    AS 
        new (repo_owner, repo_name, number, title, body, labels)
    WHERE EXISTS (
        SELECT (owner, name) FROM repos WHERE (repos.owner, repos.name) = (new.repo_owner, new.repo_name)
                 )
    ON CONFLICT ON CONSTRAINT unique_issues DO 
    UPDATE SET 
        title = EXCLUDED.title
      , body = EXCLUDED.body
      , labels = EXCLUDED.labels;
```
Or this?

Reply:
@rashadg1030 I've updated your last comment with formatting I think okay. Formatting SQL properly is a difficult task in case of complicated queries... Should be fine for now.